android/capacitor/src/main/java/com/getcapacitor/plugin/util/CapacitorHttpUrlConnection.java

```diff
@@ -262,7 +262,7 @@ private void writeObjectRequestBody(JSObject object) throws IOException, JSONExc
 
     private void writeFormDataRequestBody(String contentType, JSArray entries) throws IOException, JSONException {
         try (DataOutputStream os = new DataOutputStream(connection.getOutputStream())) {
-            String boundary = contentType.split(";")[1].split("=")[1];
+            String boundary = extractBoundary(contentType);
```

Comment:
Problem here is that we do not have a fallback as with iOS to set a random UUID in case no explicit boundary was already provided. I am kind of wondering how / if this has worked before with multipart/form-data in that case :thinking:.

Reply:
Resolved via 459e60c

This should be a more robust solution and be compatible with how expectations are in web apis (fetch / xhr) where the browser automatically adds the boundary when it's not present.